packages/debug/src/index.ts

```diff
@@ -1,88 +1,200 @@
-import debug from 'debug'
+/* eslint-disable no-var */
+import * as kleur from 'kleur/colors'
+import { bold } from 'kleur/colors'
 
-import { Debug, Debugger } from './types'
+const MAX_ARGS_HISTORY = 100
+const COLORS = ['green', 'yellow', 'blue', 'magenta', 'cyan', 'red']
 
-const MAX_LOGS = 100
+const argsHistory: [namespace: string, ...unknown[]][] = []
+let lastTimestamp = Date.now()
+let lastColor = 0
 
-const debugArgsHistory: any[] = []
+globalThis.DEBUG ??= process.env.DEBUG ?? ''
+globalThis.DEBUG_COLORS ??= process.env.DEBUG_COLORS ? process.env.DEBUG_COLORS === 'true' : true
 
-// Patch the Node.js logger to use `console.debug` or `console.log` (similar to
-// the browser logger) in the Edge Client.
-if (typeof process !== 'undefined' && typeof process.stderr?.write !== 'function') {
-  debug.log = console.debug ?? console.log
+/**
+ * Top-level utilities to configure the debug module.
+ *
+ * @example
+ * ```ts
+ * import Debug from '@prisma/debug'
+ * Debug.enable('prisma:client')
+ * const debug = Debug('prisma:client')
+ * debug('Hello World')
+ * ```
+ */
+const topProps = {
+  enable(namespace: any) {
+    if (typeof namespace === 'string') {
+      globalThis.DEBUG = namespace
+    }
+  },
+  disable() {
+    const prev = globalThis.DEBUG
+    globalThis.DEBUG = ''
+    return prev
+  },
+  // this is the core logic to check if logging should happen or not
+  enabled(namespace: string) {
+    // these are the namespaces that we are listening to in DEBUG=...
+    const listenedNamespaces: string[] = globalThis.DEBUG.split(',').map((s: string) => {
+      return s.replace(/[.+?^${}()|[\]\\]/g, '\\$&') // escape regex except "*"
+    })
+
+    // we take incoming namespaces and check then against listened
+    const isListened = listenedNamespaces.some((listenedNamespace) => {
+      if (listenedNamespace === '' || listenedNamespace[0] === '-') return false
+
+      return namespace.match(RegExp(listenedNamespace.split('*').join('.*') + '$'))
+    })
+
+    // we take incoming namespaces and check then against excluded
+    const isExcluded = listenedNamespaces.some((listenedNamespace) => {
+      if (listenedNamespace === '' || listenedNamespace[0] !== '-') return false
+
+      return namespace.match(RegExp(listenedNamespace.slice(1).split('*').join('.*') + '$'))
+    })
+
+    return isListened && !isExcluded
+  },
+  log: (...args: string[]) => {
+    const [namespace, format, ...rest] = args
+    let logger: (...args: unknown[]) => void
+
+    if (
+      typeof require === 'function' &&
+      typeof process !== 'undefined' &&
+      typeof process.stderr !== 'undefined' &&
+      typeof process.stderr.write === 'function'
+    ) {
+      logger = (...args: unknown[]) => {
+        const util = require(`${'util'}`)
+        process.stderr.write(util.format(...args) + '\n')
+      }
+    } else {
+      logger = console.warn ?? console.log
```

Comment:
```suggestion
      logger = console.warn ?? console.log
```
So if warn does not exist it would fallback to log, do you know a case where this would happen? Just curious.

Reply:
Honestly, I don't think it would ever happen

Reply:
I also thought so